plugin.js

```diff
@@ -145,10 +145,20 @@ class CspHtmlWebpackPlugin {
   apply(compiler) {
     if (compiler.hooks) {
       compiler.hooks.compilation.tap('CspHtmlWebpackPlugin', compilation => {
-        compilation.hooks.htmlWebpackPluginAfterHtmlProcessing.tapAsync(
-          'CspHtmlWebpackPlugin',
-          this.processCsp.bind(this)
-        );
+        // HtmlWebPackPlugin 3.x
+        if (compilation.hooks.htmlWebpackPluginAfterHtmlProcessing) {
+          compilation.hooks.htmlWebpackPluginAfterHtmlProcessing.tapAsync(
+            'CspHtmlWebpackPlugin',
+            this.processCsp.bind(this)
+          );
+        } else {
+          // HtmlWebPackPlugin 4.x
+          const HtmlWebpackPlugin = require('html-webpack-plugin'); // eslint-disable-line global-require
```

Comment:
Is this the recommended way to load html webpack plugin?

I see the subresource integrity plugin using a slightly different method which might be preferable: https://github.com/waysact/webpack-subresource-integrity/blob/master/index.js#L18-L25

Reply:
Since the plugin directly utilise `html-webpack-plugin` I think this is sufficient. Based on https://github.com/jantimon/html-webpack-plugin/blob/webpack-4/README.md#afteremit-hook

Reply:
That is also because this library supports older version of webpack and html-webpack-plugin. We don't want to import anything until needed.